From b09aaa25b54fbfc1c3ecd0495c922114ca6582de Mon Sep 17 00:00:00 2001
From: Zeng Zhaoming <b32542@freescale.com>
Date: Tue, 9 Aug 2011 02:35:26 +0800
Subject: [PATCH 0332/2463] ENGR00154526 MX5x, SSI: audio capture not
 supported

Audio capture not support in 2.6.38 kernel, it is caused
by not setting ssi correctly in clock and sync method.

Signed-off-by: Zeng Zhaoming <b32542@freescale.com>
---
 arch/arm/mach-mx5/board-mx51_babbage.c |    2 +-
 arch/arm/mach-mx5/board-mx53_loco.c    |    2 +-
 arch/arm/mach-mx5/board-mx53_smd.c     |    2 +-
 3 files changed, 3 insertions(+), 3 deletions(-)

diff --git a/arch/arm/mach-mx5/board-mx51_babbage.c b/arch/arm/mach-mx5/board-mx51_babbage.c
index 251c6ec..1b5614f 100755
--- a/arch/arm/mach-mx5/board-mx51_babbage.c
+++ b/arch/arm/mach-mx5/board-mx51_babbage.c
@@ -587,7 +587,7 @@ static void __init fixup_mxc_board(struct machine_desc *desc, struct tag *tags,
 }
 
 static struct imx_ssi_platform_data bbg_ssi_pdata = {
-	.flags = IMX_SSI_DMA,
+	.flags = IMX_SSI_DMA | IMX_SSI_SYN,
 };
 
 extern int mx51_babbage_init_mc13892(void);
diff --git a/arch/arm/mach-mx5/board-mx53_loco.c b/arch/arm/mach-mx5/board-mx53_loco.c
index 34ec414..e594c85 100755
--- a/arch/arm/mach-mx5/board-mx53_loco.c
+++ b/arch/arm/mach-mx5/board-mx53_loco.c
@@ -533,7 +533,7 @@ static int loco_sgtl5000_init(void)
 }
 
 static struct imx_ssi_platform_data loco_ssi_pdata = {
-	.flags = IMX_SSI_DMA,
+	.flags = IMX_SSI_DMA | IMX_SSI_SYN,
 };
 
 static struct mxc_audio_platform_data loco_audio_data = {
diff --git a/arch/arm/mach-mx5/board-mx53_smd.c b/arch/arm/mach-mx5/board-mx53_smd.c
index 179464b..04db315 100755
--- a/arch/arm/mach-mx5/board-mx53_smd.c
+++ b/arch/arm/mach-mx5/board-mx53_smd.c
@@ -675,7 +675,7 @@ static struct platform_device smd_audio_device = {
 };
 
 static struct imx_ssi_platform_data smd_ssi_pdata = {
-	.flags = IMX_SSI_DMA,
+	.flags = IMX_SSI_DMA | IMX_SSI_SYN,
 };
 
 static struct fsl_mxc_lcd_platform_data lcdif_data = {
-- 
1.7.7.4

